collect_app/src/main/java/org/odk/collect/android/widgets/ExVideoWidget.java

```diff
@@ -0,0 +1,148 @@
+    private final ExternalAppIntentProvider externalAppIntentProvider;
+    private final ActivityAvailability activityAvailability;
+
+    File answerFile;
+
+    public ExVideoWidget(Context context, QuestionDetails questionDetails, QuestionMediaManager questionMediaManager,
+                         WaitingForDataRegistry waitingForDataRegistry, MediaUtils mediaUtils,
+                         ExternalAppIntentProvider externalAppIntentProvider, ActivityAvailability activityAvailability) {
+        super(context, questionDetails);
+
+        this.waitingForDataRegistry = waitingForDataRegistry;
+        this.questionMediaManager = questionMediaManager;
+        this.mediaUtils = mediaUtils;
+        this.externalAppIntentProvider = externalAppIntentProvider;
+        this.activityAvailability = activityAvailability;
+    }
+
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        setupAnswerFile(prompt.getAnswerText());
+
+        binding = ExVideoWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+
+        binding.captureVideoButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+        binding.playVideoButton.setTextSize(TypedValue.COMPLEX_UNIT_DIP, answerFontSize);
+        binding.captureVideoButton.setVisibility(questionDetails.isReadOnly() ? GONE : VISIBLE);
+        binding.captureVideoButton.setOnClickListener(view -> launchExternalApp());
+        binding.playVideoButton.setOnClickListener(view -> mediaUtils.openFile(getContext(), answerFile, "video/*"));
+        binding.playVideoButton.setEnabled(answerFile != null);
+
+        return binding.getRoot();
+    }
+
+    @Override
+    public void deleteFile() {
+        questionMediaManager.deleteAnswerFile(getFormEntryPrompt().getIndex().toString(), answerFile.getAbsolutePath());
+        answerFile = null;
+    }
+
+    @Override
+    public void clearAnswer() {
+        deleteFile();
+        binding.playVideoButton.setEnabled(false);
+        widgetValueChanged();
+    }
+
+    @Override
+    public IAnswerData getAnswer() {
+        return answerFile != null ? new StringData(answerFile.getName()) : null;
+    }
+
+    @Override
+    public void setData(Object object) {
+        if (answerFile != null) {
+            clearAnswer();
+        }
+
+        if (object instanceof File && mediaUtils.isVideoFile((File) object)) {
+            answerFile = (File) object;
+            if (answerFile.exists()) {
+                questionMediaManager.replaceAnswerFile(getFormEntryPrompt().getIndex().toString(), answerFile.getAbsolutePath());
+                binding.playVideoButton.setEnabled(true);
+                widgetValueChanged();
+            } else {
+                Timber.e("Inserting Video file FAILED");
+            }
+        } else if (object != null) {
+            Timber.e("ExVideoWidget's setBinaryData must receive a video file but received: %s", object.getClass());
```

Comment:
I think if we do want to log something here we should log the MIME type. I don't really know if that's actionable on our end, though. It probably just means that a form called out to an app that can return files other than videos.

I don't think the user gets any feedback, though? Could we do a toast that says something like "Application returned an invalid file type"?

Reply:
Ok I added it.